apstools/beamtime/bss_info.py

```diff
@@ -0,0 +1,459 @@
+                break
+
+    if len(proposal_id) > 0:
+        proposal = getProposal(proposal_id, cycle, beamline)
+        bss.proposal.mail_in_flag.put(proposal["mailInFlag"] in ("Y", "y"))
+        bss.proposal.proprietary_flag.put(proposal["proprietaryFlag"] in ("Y", "y"))
+        bss.proposal.submitted_date.put(proposal["submittedDate"])
+        bss.proposal.title.put(proposal["title"])
+
+        bss.proposal.user_last_names.put(
+            ",".join([user["lastName"] for user in proposal["experimenters"]])
+        )
+        bss.proposal.user_badges.put(
+            ",".join([user["badge"] for user in proposal["experimenters"]])
+        )
+        for i, user in enumerate(proposal["experimenters"]):
+            obj = getattr(bss.proposal, f"user{i+1}")
+            obj.badge_number.put(user["badge"])
+            obj.email.put(user["email"])
+            obj.first_name.put(user["firstName"])
+            obj.last_name.put(user["lastName"])
+            obj.institution.put(user["institution"])
+            obj.institution_id.put(str(user["instId"]))
+            obj.user_id.put(str(user["id"]))
+            obj.pi_flag.put(user.get("piFlag") in ("Y", "y"))
+            if i == 9:
+                break
+
+
+def epicsSetup(prefix, beamline, cycle=None):
+    if beamline not in listAllBeamlines():
+        raise ValueError(f"{beamline} is not known")
+    if cycle not in listAllRuns():
+        raise ValueError(f"{cycle} is not known")
+
+    bss = connect_epics(prefix)
+
+    cycle = cycle or getCurrentCycle()
+    sector = int(beamline.split("-")[0])
+    print(f"setup EPICS {prefix} {beamline} cycle={cycle} sector={sector}")
+
+    bss.clear()
+    bss.esaf.aps_cycle.put(cycle)
+    bss.proposal.beamline_name.put(beamline)
+    bss.esaf.sector.put(str(sector))
+
+
+def getCurrentCycle():
+    return api_bss.getCurrentRun()["name"]
+
+
+def getCurrentEsafs(sector):
+    if isinstance(sector, int):
+        sector = f"{sector:02d}"
+    if len(sector) == 1:
+        sector = "0" + sector
+    tNow = datetime.datetime.now()
+    esafs = api_esaf.listEsafs(sector=sector, year=tNow.year)
+    results = []
+    for esaf in esafs:
+        if tNow < iso2datetime(esaf["experimentStartDate"]):
+            continue
+        if tNow > iso2datetime(esaf["experimentEndDate"]):
+            continue
+        results.append(esaf)
+    return results
+
+
+def getCurrentInfo(beamline):
+    sector = beamline.split("-")[0]
+    tNow = datetime.datetime.now()
+
+    matches = []
+    for esaf in api_esaf.listEsafs(sector=sector, year=tNow.year):
+        print(f"ESAF {esaf['esafId']}: {esaf['esafTitle']}")
+        esaf_badges = [user["badge"] for user in esaf["experimentUsers"]]
+        for run in listRecentRuns():
+            for proposal in api_bss.listProposals(beamlineName=beamline, runName=run):
+                print(f"proposal {proposal['id']}: {proposal['title']}")
+                proposal["experimenters"]
+                user["badge"]
```

Comment:
Codacy found an issue: [Statement seems to have no effect](https://app.codacy.com/gh/BCDA-APS/apstools/pullRequest?prid=5764884)